pkg/apis/hive/v1/gcp/machinepools.go

```diff
@@ -9,6 +9,20 @@ type MachinePool struct {
 	// eg. n1-standard-4
 	InstanceType string `json:"type"`
 
+	// ControlPlaneSubnet defines the subnet where the control plane will be deployed.
+	// The value should be the name of the subnet.
+	// If ommitted one will be created by the installer.
+	ControlPlaneSubnet string `json:"control_plane_subnet,omitempty"`
+
+	// ComputeSubnet defines the subnet where the compute nodes will be deployed.
+	// The value should be the name of the subnet.
+	// If ommitted one will be created by the installer.
+	ComputeSubnet string `json:"compute_subnet,omitempty"`
```

Comment:
What is the goal of letting the user define the control plane and compute subnet separately? seems like the user should only care which subnet this machinepool should be deployed..

From what i can see MachinePools today are mostly used to manage the `compute` role in cluster, (infra is still compute with additional stipulation) ... So maybe we treat the subnet as installertypes.platform.gcp.computeSubnet .

Reply:
> What is the goal of letting the user define the control plane and compute subnet separately? seems like the user should only care which subnet this machinepool should be deployed..

I was just exposing the API to the user as I saw the installer allowed for it. Alas, as you've mentioned - this _currently_ doesn't have a use case

> From what i can see MachinePools today are mostly used to manage the compute role in cluster, (infra is still compute with additional stipulation) ... So maybe we treat the subnet as installertypes.platform.gcp.computeSubnet .

So leave out `ControlPlaneSubnet` :+1: